docs/api/tls.md

```diff
@@ -0,0 +1,110 @@
+Pi-hole creates a self-signed certificate during installation. This certificate is used to encrypt the web interface and the API. While this certificate is secure, it is not trusted by your browser. This means that you will get a warning when you open the web interface or use the API like:
+
+![Warning in Firefox](../images/api/firefox-tls-insecure.png)
+
+You can avoid this warning by either using your own secure certificate and domain or by adding the certificate to your trusted certificates.
+
+## Adding the CA to your browser
+
+If you want to use the self-signed certificate, you can add the automatically generated certificate authority (CA) to your browser. This will make your browser trust all certificates signed by this CA. This is the same mechanism that is used by your browser to trust certificates signed by a certificate authority like Let's Encrypt. The following instructions show how to add the CA to Firefox and Chrome. The instructions for other browsers are similar.
+
+It is also possible to add the CA to your operating system's certificate store. This will make all applications that use the operating system's certificate store trust the CA. The instructions for this are operating system specific and are not covered here.
+
+Note that you have to add the **CA** certificate (e.g., `/etc/pihole/tls_ca.crt`) and not the server certificate (e.g., `/etc/pihole/tls.pem`).
+
+It is worth noting that the certificate is only valid for the domain that you had configured during certificate creation. If you access the Pi-hole web interface using a different domain, you will get a warning. This is because the certificate does not match the domain. You can either add the certificate for the other domain as well or you can create a new certificate for the other domain.
+
+<!-- markdownlint-disable code-block-style -->
+!!! warning "Security warning"
```

Comment:
Please add some instruction how to generate/re-generate a new certificate (e.g delete the old and restart pihole-FTL or using `pihole-FTL --gen-x509 /etc/pihole/tls.pem`)

Reply:
done